uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/absence-tab/components/annual-entitlements_test.js

```diff
@@ -0,0 +1,136 @@
+
+    beforeEach(module('leave-absences.templates', 'absence-tab', 'common.mocks', 'leave-absences.mocks', function (_$provide_) {
+      $provide = _$provide_;
+    }));
+
+    beforeEach(inject(function (AbsenceTypeAPIMock, AbsencePeriodAPIMock, EntitlementAPIMock) {
+      $provide.value('AbsenceTypeAPI', AbsenceTypeAPIMock);
+      $provide.value('AbsencePeriodAPI', AbsencePeriodAPIMock);
+      $provide.value('EntitlementAPI', EntitlementAPIMock);
+    }));
+
+    beforeEach(inject(['api.contact.mock', 'AbsenceTypeAPIMock', 'AbsencePeriodAPIMock', 'EntitlementAPIMock',
+      function (_ContactAPIMock_, _AbsenceTypeAPIMock_, _AbsencePeriodAPIMock_, _EntitlementAPIMock_) {
+        $provide.value('api.contact', _ContactAPIMock_);
+
+        ContactAPIMock = _ContactAPIMock_;
+        AbsenceTypeAPIMock = _AbsenceTypeAPIMock_;
+        AbsencePeriodAPIMock = _AbsencePeriodAPIMock_;
+        EntitlementAPIMock = _EntitlementAPIMock_;
+      }]));
+
+    beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _Contact_, _AbsenceType_, _AbsencePeriod_, _Entitlement_) {
+      $compile = _$compile_;
+      $log = _$log_;
+      $rootScope = _$rootScope_;
+
+      spyOn($log, 'debug');
+
+      compileComponent();
+    }));
+
+    it('is initialized', function () {
+      expect($log.debug).toHaveBeenCalled();
+    });
+
+    it('has a contact to load for', function () {
+      expect(controller.contactId).toEqual(contactId);
+    });
+
+    it('has absence types', function () {
+      expect(JSON.parse(angular.toJson(controller.absenceTypes))).toEqual(AbsenceTypeAPIMock.mockedAbsenceTypes());
+    });
+
+    it('has absence periods', function () {
+      expect(controller.absencePeriods).toEqual(jasmine.any(Array));
+    });
+
+    it('has loaded absence periods', function () {
+      expect(controller.loading.absencePeriods).toEqual(false);
+    });
+
+    it('navigates to a correct edit entitlements page', function () {
+      expect(controller.editEntitlementsPageUrl).toEqual('/index.php?q=civicrm/admin/leaveandabsences/periods/manage_entitlements&cid=' + contactId + '&returnUrl=%2Findex.php%3Fq%3Dcivicrm%2Fcontact%2Fview%26cid%3D202%26selectedChild%3Dabsence');
+    });
+
+    describe('absence period', function () {
+      var absencePeriod, mockedAbsencePeriod, mockedEntitlements;
+
+      beforeEach(function () {
+        absencePeriod = controller.absencePeriods[0];
+        mockedAbsencePeriod = _.sortBy(AbsencePeriodAPIMock.mockedAbsencePeriods(), function (absencePeriod) {
+          return -moment(absencePeriod.start_date).valueOf();
+        })[0];
+        mockedEntitlements = EntitlementAPIMock.mockedEntitlements();
+      });
+
+      it('has period', function () {
+        expect(absencePeriod.period).toEqual(moment(mockedAbsencePeriod.start_date).format('YYYY'));
+      });
+
+      it('has entitlements', function () {
+        expect(absencePeriod.entitlements).toEqual(jasmine.any(Array));
```

Comment:
I think this is what confused you:
`describe('absence', function () {`
I now changed it to: 
`describe('entitlement', function () {`
As you can see I test the actual data later